server/controller/program.controller.js

```diff
@@ -0,0 +1,434 @@
+import express from 'express';
+
+import passport from 'passport';
+
+import { validationResult } from 'express-validator';
+import {
+  validateCreateDegreeProgram,
+  validateFetchDegreeProgram,
+  validateUpdateDegreeProgramByParam,
+  validateUpdateDegreeProgramById,
+  validateDeleteDegreeProgram,
+  validateObjStructure,
+} from './validation/program.validation';
+
+import {
+  ID_OR_ANY_OF_THREE_PARAMETERS_IS_REQUIRED,
+  DEGREE_PROGRAM_ALREADY_EXISTS,
+  DEGREE_PROGRAM_NOT_FOUND,
+  SERVER_ERROR,
+  ID_OR_ALL_THREE_OTHER_PARAMETERS_IS_REQUIRED,
+} from './constant';
+
+import { Program } from '../database/models';
+
+const programController = express.Router();
+
+/**
+ * @route   POST /program/create/
+ * @desc    Creates a new degree program
+ * @access  Private
+ */
+programController.post(
+  '/create',
```

Comment:
usually, the standard for POST to create an object doesn't need to have `/create`, we can just use `'/'` -> Developers will know it's POST/ for create an object.